galley/pkg/config/analysis/analyzers/testdata/virtualservice_host_not_found_gateway.yaml

```diff
@@ -0,0 +1,175 @@
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: testing-gateway-01-test-01
+  namespace: istio-system
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - hosts:
+    - testing-01.com
+    port:
+      name: http
+      number: 80
+      protocol: HTTP
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: testing-service-01-test-01
+  namespace: default
+spec:
+  gateways:
+  - istio-system/testing-gateway-01-test-01
+  hosts:
+  - testing-01.com # Expected: no validation error since this host exists
+  http:
+  - match:
+    - uri:
+        prefix: /
+    route:
+    - destination:
+        host: ratings
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: testing-service-02-test-01
+  namespace: default
+spec:
+  gateways:
+  - istio-system/testing-gateway-01-test-01
+  hosts:
+  - wrong-01.com # Expected: validation error since this host does not exist
+  http:
+  - match:
+    - uri:
+        prefix: /
+    route:
+    - destination:
+        host: ratings
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: testing-gateway-01-test-02
+  namespace: istio-system
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - hosts:
+    - '*.testing-02.com'
+    port:
+      name: http
+      number: 80
+      protocol: HTTP
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: testing-service-01-test-02
+  namespace: default
+spec:
+  gateways:
+  - istio-system/testing-gateway-01-test-02
+  hosts:
+  - 'web.testing-02.com' # Expected: no validation error since this host match the wildcard
+  http:
+  - match:
+    - uri:
+        prefix: /
+    route:
+    - destination:
+        host: ratings
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: testing-service-02-test-02
+  namespace: default
+spec:
+  gateways:
+  - istio-system/testing-gateway-01-test-02
+  hosts:
+  - 'web.wrong.com' # Expected: validation error since this host does not exist
+  http:
+  - match:
+    - uri:
+        prefix: /
+    route:
+    - destination:
+        host: ratings
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: testing-gateway-01-test-03
+  namespace: istio-system
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
```

Comment:
we may also want a test with multiple `servers` in a single gateway

Reply:
let me add another test

Reply:
adding this test with test suffix `*-test-04`